packages/vc-verification/src/Ipfs-credential-resolver.ts

```diff
@@ -0,0 +1,77 @@
+import { providers, utils } from 'ethers';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import { IDidStore } from '@ew-did-registry/did-store-interface';
+import { OffChainClaim } from './models';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import * as jwt from 'jsonwebtoken';
+import { upgradeChainId } from './upgrade-chainid';
+import { CredentialResolver } from './credential-resolver';
+
+export class IpfsCredentialResolver implements CredentialResolver {
+  private _ipfsStore: IDidStore;
+  private _resolver: Resolver;
+
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    didStore: DidStore
+  ) {
+    this._ipfsStore = didStore;
+    this._resolver = new Resolver(provider, registrySetting);
+  }
+
+  /**
+   *
+   * @param serviceEndpoint
+   * @returns {string}
+   */
+  async getCredential(did: string, role: string) {
+    const offChainClaims = await this.offchainClaimsOf(did);
+    return offChainClaims.find(
+      (claim) =>
+        claim.claimType === role || utils.namehash(claim.claimType) === role
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async isOffchainClaim(claim: any): Promise<boolean> {
```

Comment:
```suggestion
  private async isOffchainClaim(claim: unknown): claim is OffChainClaim {
```
What do you think of that?

Reply:
This might throw error, reason being the method is async and also the `Object` property doesn't handle unknown type I guess.

Reply:
Ohh yeah, type guard doesn't support promises yet :/
```suggestion
  private async isOffchainClaim(claim: unknown): Promise<boolean> {
    if (!claim) return false;
    if (typeof claim !== 'object') return false;
    const offChainClaimProps = [
      'claimType',
      'claimTypeVersion',
      'issuedToken',
      'iss',
    ];
    const claimProps = Object.keys(claim);
    return offChainClaimProps.every((p) => claimProps.includes(p));
  }
```

Reply:
But we can still use `unknown` type and it's highly recommended pattern in typescript projects to avoid `any` and use `unknown` instead

Reply:
https://stackoverflow.com/questions/51439843/unknown-vs-any
https://dev.to/arikaturika/typescript-why-to-use-unknown-instead-of-any-41i8

Reply:
Yeah, agreed!